xdsl/dialects/builtin.py

```diff
@@ -199,11 +205,11 @@ class SymbolRefAttr(ParametrizedAttribute):
     def __init__(
         self,
         root: str | StringAttr,
-        nested: list[str] | list[StringAttr] | ArrayAttr[StringAttr] = [],
+        nested: Sequence[str] | Sequence[StringAttr] | ArrayAttr[StringAttr] = [],
     ) -> None:
         if isinstance(root, str):
             root = StringAttr(root)
-        if isinstance(nested, list):
+        if isinstance(nested, Sequence):
```

Comment:
```suggestion
        if not isinstance(nested, ArrayAttr):
```
I'm not sure how exactly Python tests for interface conformance but it seems that testing for the target class directly is a bit simpler